### PR TITLE
sql/index/pilosa: upgrade to go-pilosa v0.10.0 and migrate API

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes https://github.com/src-d/go-mysql-server/issues/353

We're up to date now 😎

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,14 +23,14 @@ services:
...
-  - docker run -d --name pilosa -p 127.0.0.1:10101:10101 pilosa/pilosa:v0.9.0
+  - docker pull pilosa/pilosa:v1.1.0
+  - docker run -d --name pilosa -p 127.0.0.1:10101:10101 pilosa/pilosa:v1.1.0
   - docker ps -a
 
 install:
   - go get -u github.com/golang/dep/cmd/dep
   - touch Gopkg.toml
-  - dep ensure -v -add "github.com/pilosa/go-pilosa@0.9.0" "github.com/pilosa/pilosa@1.0.2" "github.com/moovweb/rubex@b3d9ff6ad7d9b14f94a91c8271cd9ad9e77132e5"
+  - dep ensure -v -add "github.com/pilosa/go-pilosa@1.1.0" "github.com/pilosa/pilosa@1.0.2" "github.com/moovweb/rubex@b3d9ff6ad7d9b14f94a91c8271cd9ad9e77132e5"
```
Comment (ajnavarro) [inline .travis.yml]: shall we move `github.com/pilosa/pilosa@1.0.2` dependency to `1.1.0` too?

File: README.md [hunk 2]
```diff
@@ -44,10 +44,10 @@ You might run into trouble using `go get` on the library because of breaking cha
 
 ```
 cd $GOPATH/src/github.com/pilosa/go-pilosa
-git checkout v0.9.0
+git checkout v0.10.0
```
Comment (ajnavarro) [inline README.md]: did you mean `v1.1.0` ?

File: go.mod [hunk 3]
```diff
@@ -3,16 +3,20 @@ module gopkg.in/src-d/go-mysql-server.v0
...
 	github.com/boltdb/bolt v1.3.1
 	github.com/cespare/xxhash v1.0.0 // indirect
+	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/gogo/protobuf v1.1.1 // indirect
 	github.com/golang/protobuf v1.2.0 // indirect
 	github.com/mitchellh/hashstructure v0.0.0-20170609045927-2bca23e0e452
+	github.com/moovweb/rubex v0.0.1
 	github.com/opentracing/opentracing-go v1.0.2
-	github.com/pilosa/go-pilosa v0.9.0
+	github.com/pilosa/go-pilosa v0.10.0
```
Comment (ajnavarro) [inline go.mod]: `1.1.0` ?

**Comment by kuba--** [comment]
There are 3 things:
go-pilosa used by pilosa driver (if we upgrade to the latest 0.10 we can use server 1.X)
pilosa package used by pilosalib (now it uses 1.0.2, but we can upgrade to the latest 1.1.2 i think)
server used in docker for pilosa driver (now it's 0.9, but if we upgrade go-pilosa we can upgrade to the latest 1.X)

**Comment by erizocosmico** [comment]
Fixed

**Pull request merged**